src/java.base/share/classes/java/util/zip/ZipFile.java

```diff
@@ -411,13 +411,10 @@ public InputStream getInputStream(ZipEntry entry) throws IOException {
                 case DEFLATED:
                     // Inflater likes a bit of slack
                     // MORE: Compute good size for inflater stream:
-                    long size = CENLEN(zsrc.cen, pos) + 2;
+                    long size = CENSIZ(zsrc.cen, pos);
                     if (size > 65536) {
                         size = 8192;
                     }
```

Comment:
Not sure if this clamping makes sense? We clamp the size at 8192, but only when size is larger than 65536?

Wondering if we should simply clamp to 8192 instead:


```suggestion
                   int size = Math.clamp(CENSIZ(zsrc.cen, pos), 2, 8192);
```

Reply:
I'm curious if it would be beneficial to increase the max clamp size to 16384 bytes similar to https://bugs.openjdk.org/browse/JDK-8299336 / https://github.com/openjdk/jdk/pull/11783

```suggestion
            int size = Math.clamp(CENSIZE(zsrc.can, pos), 2, 16384);
```

Reply:
Yes, there may be benefits in increasing this max buffer size for larger entries, but I think that requires a more extensive analysis and discussion than what want to do in this PR.

99 percent of classes in guava.jar have a compressed size less than ~6700 bytes. The average compressed size is ~1300 bytes.

So this does not seem to matter for class loading, which is probably what ZipFile is most optimized for.